bin/tronfig

```diff
@@ -103,51 +82,81 @@ def delete_config(client, config_name):
     raise SystemExit("tronfig deletion failed")
 
 
-def update_from_stdin(client, config_name, check=False):
-    config_content = sys.stdin.read()
-    config_hash = client.config(config_name)['hash']
-    if validate(config_name, config_content):
-        if upload_config(
-                client, config_name, config_content, config_hash, check=check):
-            return
-    raise SystemExit("tronfig failed")
+def validate_dir(path):
+    try:
+        manifest_dir = tempfile.mkdtemp()
+        manifest = manager.ManifestFile(manifest_dir)
+        manifest.create()
+        for fname in os.listdir(path):
+            if fname.endswith(".yaml"):
+                namespace = fname[0:-5]
+                manifest.add(namespace, os.path.join(path, fname))
+
+        config_manager = manager.ConfigManager(path, manifest)
+        config_manager.load()
+    except ConfigError, e:
+        traceback.print_exc()
+        return str(e)
+    finally:
+        if manifest_dir:
+            shutil.rmtree(manifest_dir)
 
 
-def update_with_editor(client, config_name):
-    if not sys.stdout.isatty():
-        raise SystemExit("No editing possible.")
+def get_config_input(namespace, source=None):
+    if source is None:
+        source = namespace
 
-    response        = client.config(config_name)
-    config_content  = response['config']
-    config_hash     = response['hash']
-    while True:
-        config_content = edit_config(config_content)
-        if not config_content:
-            log.warn("Cancelling edit.")
-            return
+    if source == '-':
+        source_io = sys.stdin
+        namespace = schema.MASTER_NAMESPACE
+    else:
+        source_io = open(source, 'r')
+        namespace = os.path.basename(source)[0:-5]
 
-        if validate(config_name, config_content):
-            if upload_config(client, config_name, config_content, config_hash):
-                return
+    content = source_io.read()
 
-        response = raw_input(
-            "There are errors in your configuration. Continue editing? (y/n): ")
-        if response[:1].lower() == 'n':
-            return
+    return namespace, content, source
```

Comment:
get_config_input still returns namespace, content, source.
There's one place below (validation) where all 3 are unpacked, but a couple others where it's assumed get_config_input just returns namespace, content.

Not sure which is correct, but should match up.